transport/peer/picker.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2013 Google Inc.
+Copyright 2024 Derrick J Wippler
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package peer
+
+import (
+	"crypto/md5"
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/segmentio/fasthash/fnv1"
+)
+
+const defaultReplicas = 50
```

Comment:
Might be worth explaining somewhere why 50 replicas is the default. I definitely don't initially grok what considerations I should keep in mind when choosing a replica size 👀